google-cloud-clients/google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1beta1/stub/ResourceHeaderTest.java

```diff
@@ -0,0 +1,168 @@
+ */
+package com.google.cloud.bigquery.storage.v1beta1.stub;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import com.google.api.gax.core.NoCredentialsProvider;
+import com.google.api.gax.grpc.testing.InProcessServer;
+import com.google.api.gax.grpc.testing.LocalChannelProvider;
+import com.google.api.gax.rpc.FixedHeaderProvider;
+import com.google.api.gax.rpc.UnimplementedException;
+import com.google.cloud.bigquery.storage.v1beta1.BigQueryStorageClient;
+import com.google.cloud.bigquery.storage.v1beta1.BigQueryStorageGrpc.BigQueryStorageImplBase;
+import com.google.cloud.bigquery.storage.v1beta1.BigQueryStorageSettings;
+import com.google.cloud.bigquery.storage.v1beta1.Storage.ReadRowsRequest;
+import com.google.cloud.bigquery.storage.v1beta1.Storage.ReadSession;
+import com.google.cloud.bigquery.storage.v1beta1.Storage.Stream;
+import com.google.cloud.bigquery.storage.v1beta1.Storage.StreamPosition;
+import com.google.cloud.bigquery.storage.v1beta1.TableReferenceProto.TableReference;
+import java.util.regex.Pattern;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ResourceHeaderTest {
+
+  private static final TableReference TEST_TABLE_REFERENCE =
+      TableReference.newBuilder()
+          .setProjectId("project")
+          .setDatasetId("dataset")
+          .setTableId("table")
+          .build();
+
+  private static final ReadSession TEST_SESSION =
+      ReadSession.newBuilder().setName("sessionName").build();
+
+  private static final Stream TEST_STREAM = Stream.newBuilder().setName("streamName").build();
+
+  private static final String NAME = "resource-header-test:123";
+
+  private static final String HEADER_NAME = "x-goog-request-params";
+
+  private static final Pattern DATASET_PATTERN =
+      Pattern.compile(".*" + "table_reference\\.dataset_id=dataset" + ".*");
+  private static final Pattern ORIGINAL_STREAM_PATTERN =
+      Pattern.compile(".*" + "original_stream\\.name=streamName" + ".*");
+  private static final Pattern PROJECT_PATTERN =
+      Pattern.compile(".*" + "table_reference\\.project_id=project" + ".*");
+  private static final Pattern READ_POSITION_PATTERN =
+      Pattern.compile(".*" + "read_position\\.stream\\.name=streamName" + ".*");
+  private static final Pattern SESSION_PATTERN =
+      Pattern.compile(".*" + "session\\.name=sessionName" + ".*");
+  private static final Pattern STREAM_PATTERN =
+      Pattern.compile(".*" + "stream\\.name=streamName" + ".*");
+
+  private static final String TEST_HEADER_NAME = "simple-header-name";
+  private static final String TEST_HEADER_VALUE = "simple-header-value";
+  private static final Pattern TEST_PATTERN = Pattern.compile(".*" + TEST_HEADER_VALUE + ".*");
+
+  private InProcessServer<?> server;
+  private LocalChannelProvider channelProvider;
+  private BigQueryStorageClient client;
+
+  @Before
```

Comment:
Please use @BeforeClass and @Afterclass below.

Reply:
This is taken verbatim from https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/ResourceHeaderTest.java -- do you want me to update that file as well?

Reply:
Scratch that -- the local channel provider stores the outgoing headers and has to be regenerated for each test.

Reply:
Moved the server setup to the class level, at least.